mmdeploy/codebase/mmocr/models/text_recognition/abi_language_decoder.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch
+
+from mmdeploy.core import FUNCTION_REWRITER
+from mmdeploy.utils import IR
+
+
+@FUNCTION_REWRITER.register_rewriter(
+    func_name='mmocr.models.textrecog.decoders.ABILanguageDecoder._get_length',
+    IR=IR.ONNX)
+def abi_language_decoder___get_length__default(self,
+                                               logit: torch.Tensor,
+                                               dim: int = -1,
+                                               **kwargs) -> torch.Tensor:
+    """Rewrite `_get_length`. Add `.float()` to cast Tensors from bool to float
+    for `cumsum` and `argmax`.
+
+    Returns the first location of padding index or the length of the entire
+    tensor otherwise.
+    """
+    # out as a boolean vector indicating the existence of end token(s)
+    out = (logit.argmax(dim=-1) == self.dictionary.end_idx)
+    abn = out.any(dim)
+    # Get the first index of end token
+    # add `.float()` to `out` for onnxruntime `cumsum()`
+    # add `.float()` before `argmax()`
+    out = ((out.float().cumsum(dim) == 1) & out).float().argmax(dim)
+    out = out + 1
+    out = torch.where(abn, out, out.clone().detach()).float()
```

Comment:
What's this `where` for?

Reply:
Just copied from MMOCR [here](https://github.com/open-mmlab/mmocr/blob/ebdf1cf90d44e278a322f327ac18eac6fff5a745/mmocr/models/textrecog/decoders/abi_language_decoder.py#L194)

Reply:
Did `out.new_tensor(logit.shape[1])==out`?
What if it did, `torch.where(abn, out, out.clone().detach())==out` whatever abn is, right?